wicket-examples/src/main/java/org/apache/wicket/examples/csp/CspApplication.java

```diff
@@ -45,9 +51,22 @@ protected void init()
 	{
 		super.init();
 
-		setHeaderResponseDecorator(response -> new ResourceAggregator(new CspNonceHeaderResponse(response, getNonce())));
+		// Decorate all header items with nonce
+		setHeaderResponseDecorator(response -> new ResourceAggregator(
+				isCspApplicable() ? new CspNonceHeaderResponse(response, getNonce()) : response
+		));
+		// add nonce to ajax response
+		getAjaxRequestTargetListeners().add((new AjaxRequestTarget.IListener()
+		{
+			@Override
+			public void onBeforeRespond(Map<String, Component> map, AjaxRequestTarget target)
+			{
+				target.addMeta("nonce", getNonce());
```

Comment:
Should this be given some special name like `wicket-nonce` or `<wicket:nonce>` ? Because `IPartialPageUpdateHandler#addMeta()` can be called by the application and it can easily override this (add another) entry. Me, as a user, I'd like to be notified somehow when my application code breaks somehow the normal framework work.

Reply:
Thanks @martin-g !

Honestly I'm not quite sure about this meta container. I was thinking about
1. placing a nonce as an attribute of `ajax-response`;
2. placing a nonce as a direct child of `ajax response`, without any kind of `meta` container;
3. adding some kind of universal container to add anything.

This one implements 3. What do you think maybe one of 1 or 2 is better?